components/admin_console/billing/billing_subscriptions/limits.tsx

```diff
@@ -272,7 +309,111 @@ const Limits = (): JSX.Element | null => {
 
                 )}
             </div>
-            <TempLaunchModalsComponent/>
+        );
+    } else if (subscriptionProduct.sku === CloudProducts.PROFESSIONAL) {
+        title = (
+            <FormattedMessage
+                id='workspace_limits.upgrade_professional'
+                defaultMessage='Professional plan data limit'
+            />
+        );
+        description = intl.formatMessage(
+            {
+                id: 'workspace_limits.upgrade_reasons.professional',
+                defaultMessage: 'On the Professional plan you can only view the most recent {fileStorage} of files. With Enterprise, experience unlimited file storage.',
+            },
+            {
+                fileStorage: asGBString(cloudLimits?.files?.total_storage || FileSizes.Gigabyte * 250, intl.formatNumber),
+            },
+        );
+        currentUsage = cloudLimits?.files?.total_storage && (
+            <SingleInlineLimit
+                name={(
+                    <FormattedMessage
+                        id='workspace_limits.file_storage'
+                        defaultMessage='File Storage'
+                    />
+                )}
+                status={(
+                    <FormattedMessage
+                        id='workspace_limits.file_storage.single_usage'
+                        defaultMessage='{actual} of {limit}'
+                        values={{
+                            actual: asGBString(usage.files.totalStorage, intl.formatNumber),
+                            limit: asGBString(cloudLimits.files.total_storage, intl.formatNumber),
+
+                        }}
+                    />
+                )}
+                percent={Math.floor((usage.files.totalStorage / cloudLimits.files.total_storage) * 100)}
+                icon='icon-folder-outline'
+            />
+        );
+    }
+
+    let panelClassname = 'ProductLimitsPanel';
+    if (singleLimitPanel) {
+        if (props.showAnnualCard) {
+            panelClassname += ' ProductLimitsPanel--left-panel';
+        } else {
+            panelClassname += ' ProductLimitsPanel--single-panel';
+        }
+    }
+    let actionsClassname = 'ProductLimitsPanel__actions';
+    if (singleLimitPanel) {
```

Comment:
nit: same in this section for the classes, those nested ifs look weird. I'd rather to use string literals when possible